### ISSUE TITLE
Plugin não funciona

### ISSUE THREAD

**Issue by fernandoplanejar** [DevelopersRede/woocommerce#1]
Ao tentar usar esse plugin a loja ficou fora do ar com o erro:

HTTP ERROR 500

Pesquisei no Google sobre o assunto e me recomendaram usar esse aqui:

https://github.com/DevelopersRede/woocommerce/releases/download/1.1.1/rede-woocommerce.zip

Com esse plugin o sistema até abriu, mas quando finalizo uma compra ocorre os seguintes erros:

– Invés de redirecionar para pagina de sucesso ele redireciona para mesma pagina anterior, sendo assim parece que a compra não foi concluida
– O pedido até entra no painel da loja, mas não envia e-mail
– O pedido não entrou no painel da eRede

Por favor você pode verificar?

Abraços!

**Comment by netojoaobatista** [comment]
@fernandoplanejar, você seguiu o procedimento de instalação?

**Comment by CarlosFarnesi** [comment]
Estou passando pelo mesmo problema:

– Invés de redirecionar para pagina de sucesso ele redireciona para mesma pagina anterior, sendo assim parece que a compra não foi concluida
– O pedido até entra no painel da loja, mas não envia e-mail
– O pedido não entrou no painel da eRede

Enviei um e-mail ao suporte deles hoje e me pediram pra enviar o arquivo json. 

Se tiverem alguma novidade, por favor, eu agradeço.

**Comment by netojoaobatista** [comment]
@CarlosFarnesi, você seguiu o procedimento de [instalação conforme instruído no README.md](https://github.com/DevelopersRede/woocommerce#instala%C3%A7%C3%A3o)?

**Comment by CarlosFarnesi** [comment]
Olá @netojoaobatista não cheguei a fazer esse procedimento. Todo ele é por linha de comando direto no servidor? Você conseguiu? Tem alguma orientação?

**Comment by CarlosFarnesi** [comment]
Olá @netojoaobatista . Sim, realizamos agora mesmo o processo todo do procedimento e continua com os mesmos problemas.

**Comment by netojoaobatista** [comment]
No `error.log` do seu sistema, vai estar escrito exatamente o motivo do erro. Tem acesso ao log?

**Comment by CarlosFarnesi** [comment]
Softdescriptor: Not enabled for this merchant.

**Comment by netojoaobatista** [comment]
Ok! Verificando sobre isso.

**Comment by netojoaobatista** [comment]
@CarlosFarnesi, segundo a [documentação do SoftDescriptor](https://www.userede.com.br/desenvolvedores/pt/produto/e-Rede#documentacao-softdescriptor): 

> Para utilizar essa funcionalidade, acesse o portal da Rede no menu e.Rede > Identificação na fatura ou entre em contato com a Central de atendimento da Rede. Caso o nome não seja cadastrado, o serviço não será habilitado.
> 
> Após a habilitação do serviço via portal, a funcionalidade será disponibilizada dentro de um prazo de até 24 horas.

**Comment by CarlosFarnesi** [comment]
@netojoaobatista Não encontro de jeito nenhum no site e.Rede (No novo e no velho) a parte onde a documentação informa sobre cadastrar sua “loja teste” informando o nome e o produto que será testado. Inclusive onde será gerado automaticamente um PV.

Com essas informações você já esta pronto para utilizar nosso playground e começar a enviar testes.

Por acaso, você sabe?

**Comment by netojoaobatista** [comment]
@CarlosFarnesi, estou verificando exatamente onde se encontra o cadastro no portal e.Rede. Enquanto encontramos a informação, por favor, apenas como teste mesmo, edite as configurações da sua loja WooCommerce e *deixe o campo Soft Descriptor* *em branco*:

![image](https://user-images.githubusercontent.com/273586/55276974-c18af100-52d9-11e9-9783-ae55d236aa16.png)

Uma vez deixado em branco, prossiga com o checkout até o fim. Veja se o problema persiste. Aqui do meu lado, tudo funcionou como esperado:

![Captura de tela de 2019-03-30 10-51-58](https://user-images.githubusercontent.com/273586/55276987-e3847380-52d9-11e9-9ce5-2f4cc343da8b.png)

**Comment by netojoaobatista** [comment]
@CarlosFarnesi, dentro do portal da Rede, procure por uma opção chamada *"nome na fatura"*; está dentro do menu e.Rede.

**Comment by netojoaobatista** [comment]
@CarlosFarnesi, especificamente sobre a loja de testes, siga os passos a seguir:

1 - Acesse [e.Rede Desenvolvedores](https://www.userede.com.br/desenvolvedores/pt/)
2 - No canto superior direito, clique em [Cadastre-se](https://www.userede.com.br/desenvolvedores/cadastro)
![Captura de tela de 2019-03-30 11-09-36](https://user-images.githubusercontent.com/273586/55277224-6a3a5000-52dc-11e9-991b-a02c0b13d7ad.png)

3 - Após preencher o cadastro - só nome, email e um captcha -, você vai ter a opção de criar uma ou mais lojas de teste. Após criar sua loja de testes, você receberá o PV e o Token, que poderão ser utilizados em ambiente de testes.

**Comment by CarlosFarnesi** [comment]
> @CarlosFarnesi, estou verificando exatamente onde se encontra o cadastro no portal e.Rede. Enquanto encontramos a informação, por favor, apenas como teste mesmo, edite as configurações da sua loja WooCommerce e _deixe o campo Soft Descriptor_ _em branco_:
> 
> ![image](https://user-images.githubusercontent.com/273586/55276974-c18af100-52d9-11e9-9783-ae55d236aa16.png)
> 
> Uma vez deixado em branco, prossiga com o checkout até o fim. Veja se o problema persiste. Aqui do meu lado, tudo funcionou como esperado:
> 
> ![Captura de tela de 2019-03-30 10-51-58](https://user-images.githubusercontent.com/273586/55276987-e3847380-52d9-11e9-9ce5-2f4cc343da8b.png)

Oi @netojoaobatista Não tenho essa opção de Soft Descript para poder deixar em branco.

**Comment by CarlosFarnesi** [comment]
> @CarlosFarnesi, dentro do portal da Rede, procure por uma opção chamada _"nome na fatura"_; está dentro do menu e.Rede.

Sim, esta com L2L e deixei da mesma maneira nas configurações do plugin.

**Comment by CarlosFarnesi** [comment]
> e.Rede Desenvolvedores

Já me cadastrei 2x e nada de receber a confirmação de cadastro por e-mail. :( Só aparece essa mensagem no site após o cadastro:

Mensagem

Parabens! Registrado com sucesso. Você deve receber em breve um e-mail para ativar sua conta. clique no link do e-mail para ativar sua conta. E, comece a criar coisas incriveis!

Mas nada de receber o e-mail para ativação da conta.

**Comment by marcjull** [comment]
@netojoaobatista
Um e-commerce que estou implementando está com um problema parecido ao relatado pelo fernandoplanejar no começo, mas no meu caso o e-mail é enviado e o log acusou sucesso diversas vezes, porém na tela a mensagem após a etapa final de compra ("finalizar compra" ou "place order" em inglês) é que há um Internal Server Error. 
Antes havia um problema de softdescriptor, para tentar de novo porque a operação não havia sido concluída. O suporte da userede me orientou a deixar o campo Nome da Fatura vazio, o que eu fiz, mas então começou o do servidor.
Houve alguma solução à parte fazer simulações, sandbox, etc? Na literatura na internet se fala muito também em debugging.

**Comment by denysbsb** [comment]
Alguém sabe da solução desse problema no plugin da Rede, que não finaliza a tela de compra mas finaliza o pedido no painel.

**Comment by marcjull** [comment]
Carlos,

Faz tanto tempo que enviei essa pergunta! Apesar disto, a resposta me
interessa, para futuras instalações do Woocommerce- no momento não tenho
nenhuma. Porém me lembro de ter conversado com isto com alguém no suporte
da Userede, que me deu essa instrução de deixar o campo Soft Descriptor em
branco, e não resolveu. A única instalação que resolveu o problema foi no
Google Cloud, nas opção Bitnami, e nela o problema não chegou a se
apresentar, a Redecard funcionava perfeitamente (a outra opção no Google
Cloud chama-se acho Google Click do Deploy, mas essa eu não cheguei a
tentar).

Obrigado, e agradeço se me avisar quando tiver informações,

Marco Juliano



Em qua., 1 de set. de 2021 às 22:02, wwwcardoso ***@***.***>
escreveu:
